### PR TITLE
DAOS-10249 rdb: Update raft

### PR DESCRIPTION
**Pull request by liw**
Update raft to fix RV assertion failures.

Signed-off-by: Li Wei <wei.g.li@intel.com>
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by kccain — DISMISSED** [review]

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+daos (2.3.103-3) unstable; urgency=medium
+  [ Li Wei ]
+  * Update raft to 0.9.2-1.403.g3d20556
+
+ -- Li Wei <wei.g.li@intel.com>  Mon, 13 Apr 2023 10:36:00 +0800
```
Comment (brianjmurrell) [inline debian/changelog:5]:
```suggestion
 -- Li Wei <wei.g.li@intel.com>  Mon, 13 Feb 2023 10:36:00 +0800
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: debian/changelog (hunk 1 above)
Comment (liw) [reply debian/changelog]: Thanks. Fixed.

**Review by kccain — DISMISSED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by liw** [comment]
Merged master to fix a conflict.

**Review by kccain — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by liw** [comment]
Merged master to fix another conflict.

**Review by kccain — APPROVED** [review]

**Comment by liw** [comment]
Merged master to fix yet another conflict.

**Pull request merged**